src/net.cpp

```diff
@@ -198,7 +232,7 @@ bool GetLocal(CService& addr, const CNetAddr *paddrPeer)
         for (const auto& entry : mapLocalHost)
         {
             int nScore = entry.second.nScore;
```

Comment:
27411: shouldn't we add
```
            // For privacy reasons, don't advertise our privacy-network address
            // to other networks and don't advertise our other-network address
            // to privacy networks.
            const Network our_net{entry.first.GetNetwork()};
            const Network peers_net{paddrPeer->GetNetwork()};
            if (our_net != peers_net &&
                (our_net == NET_ONION || our_net == NET_I2P ||
                 peers_net == NET_ONION || peers_net == NET_I2P)) {
                continue;
            }
 ```
here as well?

Reply:
Since there's only _one_ usage `GetLocal(CService&, CNetAddr*)` and it's testing against a well-known address on clearnet, this doesn't seem necessary _but_, down the road, if we expand the networks on which running masternodes are permitted, this change would be valuable. 

Have moved it to `masternode/node.cpp` in latest push so it doesn't get reused anywhere else. Resolved in latest push.